iree/base/dynamic_library_posix.cc

```diff
@@ -45,9 +45,8 @@ class DynamicLibraryPosix : public DynamicLibrary {
             new DynamicLibraryPosix(search_file_names[i], library));
       }
     }
-
     return UnavailableErrorBuilder(IREE_LOC)
-           << "Unable to open dynamic library, not found on search paths";
+           << "Unable to open dynamic library:" << dlerror();
```

Comment:
Would a space help between the colon and the error message? You could also wrap it in quotes, depending on how it is formatted:
```suggestion
           << "Unable to open dynamic library: '" << dlerror() << "'";
```

Reply:
Done! Here how it look now
```
Unable to open dynamic library:'dlopen failed: library "libc++_shared.so" not found'
```